v2/config.go

```diff
@@ -178,7 +178,7 @@ type Config struct {
 // instantiate a fully functional [DHT] client. All fields that are nil require
 // some additional information to instantiate. The default values for these
 // fields come from separate top-level methods prefixed with Default.
-func DefaultConfig() *Config {
+func DefaultConfig() (*Config, error) {
```

Comment:
I guess this is a result from a merge. We don't need an error return value here. This also affects the tests in `TestConfig_Validate`

Reply:
I've removed in the next PR in the chain https://github.com/libp2p/go-libp2p-kad-dht/pull/887

Reply:
Will remove here though - didn't realise there were so many